app/logic/validator_score_v1_logic.rb

```diff
@@ -81,10 +80,18 @@ def block_vote_history_get
         batch_uuid: p.payload[:batch_uuid]
       ).where({ account: p.payload[:validators].map(&:account) }).to_a
 
+      vote_histories = VoteAccountHistory.joins(:vote_account).where(
+        network: p.payload[:network],
+        batch_uuid: p.payload[:batch_uuid]
+      ).where('vote_account.validator_id': p.payload[:validators].pluck(:id)).to_a
```

Comment:
Nice! This is probably more performant than the similar query I had for Skipped Slot %